src/core/hle/D3D8/Direct3D9/CxbxPixelShaderTemplate.hlsl

```diff
@@ -334,6 +336,34 @@ float3 DoBumpEnv(const float4 TexCoord, const float4 BumpEnvMat, const float4 sr
 
 PS_OUTPUT main(const PS_INPUT xIn)
 {
+	//Fog
```

Comment:
Doesn't the exact same code occur in the fixed function shader hlsl too? Consider move all shared code into a single hlsli include file

Reply:
FF already has it moved to the hlsli file, however I was not sure I could include that file in the normal shader template since it may reset some variables based on FF state, I could remove it from the FF hlsli and create a new one just for fog to inclide in both templates, but it seemed a waste to include a file with 4 lines....

Reply:
Agreed. Maybe later, when more symbols get to be shared between shader paths